File: AutomaticTimetableGeneratorSystem-main/timetable-frontend/src/services/classroomService.js
import api from "./api";

const BASE_URL = "/classrooms";

const classroomService = {

  getAllClassrooms: async () => {
    const response = await api.get(BASE_URL);
    return response.data;
  },

  getClassroomById: async (id) => {
    const response = await api.get(`${BASE_URL}/${id}`);
    return response.data;
  },

  createClassroom: async (data) => {
    const response = await api.post(BASE_URL, data);
    return response.data;
  },

  updateClassroom: async (id, data) => {
    const response = await api.put(`${BASE_URL}/${id}`, data);
    return response.data;
  },

  deleteClassroom: async (id) => {
    const response = await api.delete(`${BASE_URL}/${id}`);
    return response.data;
  },

  getByRoomNumber: async (roomNumber) => {
    const response = await api.get(`${BASE_URL}/room/${roomNumber}`);
    return response.data;
  }

};

export default classroomService;

File: AutomaticTimetableGeneratorSystem-main/timetable-backend/src/main/java/com/timetable/service/ClassroomService.java
package com.timetable.service;

import java.util.List;

import com.timetable.dto.classroom.ClassroomRequest;
import com.timetable.dto.classroom.ClassroomResponse;

public interface ClassroomService {

    /**
     * Save Classroom
     */
    ClassroomResponse saveClassroom(ClassroomRequest request);

    /**
     * Update Classroom
     */
    ClassroomResponse updateClassroom(Long classroomId, ClassroomRequest request);

    /**
     * Delete Classroom
     */
    void deleteClassroom(Long classroomId);

    /**
     * Get Classroom By ID
     */
    ClassroomResponse getClassroomById(Long classroomId);

    /**
     * Get All Classrooms
     */
    List<ClassroomResponse> getAllClassrooms();

    /**
     * Get Classroom By Room Number
     */
    ClassroomResponse getClassroomByRoomNumber(String roomNumber);
}